modules/libcom/test/Makefile

```diff
@@ -288,4 +288,10 @@ TESTPROD_HOST += cvtFastPerform
 cvtFastPerform_SRCS += cvtFastPerform.cpp
 testHarness_SRCS += cvtFastPerform.cpp
 
+TESTPROD_Linux += nonEpicsThreadPriorityTest
```

Comment:
The name `nonEpicsThreadPriorityTest` doesn't make it obvious that this test code is for Posix architectures only so a name-change would be nice. The suggestions below have not been tested.

This conditional in the `Makefile` should be sufficient to make it build only on Posix targets:
```makefile
ifeq ($(POSIX),YES)
TESTPROD_HOST += nonEpicsThreadPriorityTest
testHarness_SRCS += nonEpicsThreadPriorityTest.cpp
epicsRunLibComTests_CFLAGS += -DPOSIX_TARGET
TESTS += nonEpicsThreadPriorityTest
endif
```

The `epicsRunLibComTests.c` also needs a couple of `#ifdef` wrappers so it only runs the `nonEpicsThreadPriorityTest` routine on Posix targets, e.g:
```c
#ifdef POSIX_TARGET
int nonEpicsThreadPriorityTest(void);
#endif
// etc.
```
There might be other/better macros that could be used instead of defining our own here, since `epicsRunLibComTests.c` already includes `epicsThread.h` and hence `osdThread.h`.

Reply:
@Andrew: I'll be happy to rename, suggestions welcome. The current name tries to express that the test exercises non-epics Threads (i.e., threads that were generated directly via the pthread API) that are 'promoted' to epics-threads.

How about `epicsThreadFromPthreadPriorityTest`?

Reply:
Re. POSIX_TARGET macro alternative: I had tried the _POSIX_THREAD_PRIORITY_SCHEDULING macro which is available via osdThread.h on posix targets -- unfortunately that doesn't work for RTEMS because RTEMS does indeed define that macro (and support pthreads) but EPICS does not use the pthread API. It seems therefore necessary to use some epics-specific device (such as the one you suggested).

Reply:
RTEMS 5.x defines that macro, but until you have Heinz' changes to support RTEMS-5 (which have not yet been merged either) the libCom routines use the old RTEMS APIs (and I didn't think they worked on RTEMS-5 any more; was I misled?). Thus this PR needs to assume that RTEMS will *not* be one of the targets running the test, but that will change when Heinz' stuff also gets merged. The RTEMS 4.9 and 4.10 builds that Travis does will never run that test.

Reply:
Unfortunately, testing for the 'POSIX' Makefile variable is not viable. RTEMS says POSIX=YES but still does not use the pthread API. So we need to figure out a way of finding platforms which use os/posix/osdThread.c -- not that easy with the current build system since any platform is free to override with it's own os/xyz/osdThread.c.

I'm thinking - maybe we should just scrap this test. It exercises a fringe-feature of posix/osdThread (the possibility to 'promote' a native thread to an 'epicsThread')
and I'm starting to spend way more time trying to find a way to *prevent* the test
from being built/executed than on anything else related to this patch. Plus, I'm pretty sure
travis doesn't run with the sufficient privilege to engage the RT scheduler, so what's the point...